### ISSUE TITLE
Not compatible with shapely-1.8.0

### ISSUE THREAD

**Issue by zxdawn** [holoviz/geoviews#533]
#### ALL software version info

bokeh: 2.4.1
geoviews: 1.9.2
cartopy: 0.20.1 and 0.19.0

#### Description of expected behavior and the observed behavior

`geoviews` can't add any projection.

#### Complete, minimal, self-contained example code that reproduces the issue

```
import geoviews.feature as gf
import geoviews as gv
from cartopy import crs

gv.extension("bokeh")

projections = [crs.RotatedPole, crs.LambertCylindrical, crs.Geostationary, 
               crs.AzimuthalEquidistant, crs.OSGB, crs.EuroPP, crs.Gnomonic,
               crs.Mollweide, crs.OSNI, crs.Miller, crs.InterruptedGoodeHomolosine,
               crs.SouthPolarStereo,  crs.Orthographic, crs.NorthPolarStereo, crs.Robinson,
               crs.LambertConformal, crs.AlbersEqualArea]

opts = dict(width=200, height=225, global_extent=True, axiswise=True)
gv.Layout([gf.coastline.relabel(group=p.__name__).opts(projection=p(), **opts) for p in projections]).cols(4)
```

#### Error
```
~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/shapely/geometry/base.py in array_interface_base(self)
    324             "removed in Shapely 2.0.",
    325             ShapelyDeprecationWarning, stacklevel=2)
--> 326         return self._array_interface_base()
    327 
    328     @property

TypeError: 'dict' object is not callable
```

<details><summary>Click to see full error log</summary>
<p>

```
TypeError                                 Traceback (most recent call last)
~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/IPython/core/formatters.py in __call__(self, obj, include, exclude)
    968 
    969             if method is not None:
--> 970                 return method(include=include, exclude=exclude)
    971             return None
    972         else:

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/core/dimension.py in _repr_mimebundle_(self, include, exclude)
   1314         combined and returned.
   1315         """
-> 1316         return Store.render(self)
   1317 
   1318 

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/core/options.py in render(cls, obj)
   1403         data, metadata = {}, {}
   1404         for hook in hooks:
-> 1405             ret = hook(obj)
   1406             if ret is None:
   1407                 continue

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/ipython/display_hooks.py in pprint_display(obj)
    280     if not ip.display_formatter.formatters['text/plain'].pprint:
    281         return None
--> 282     return display(obj, raw_output=True)
    283 
    284 

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/ipython/display_hooks.py in display(obj, raw_output, **kwargs)
    253     elif isinstance(obj, (Layout, NdLayout, AdjointLayout)):
    254         with option_state(obj):
--> 255             output = layout_display(obj)
    256     elif isinstance(obj, (HoloMap, DynamicMap)):
    257         with option_state(obj):

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/ipython/display_hooks.py in wrapped(element)
    144         try:
    145             max_frames = OutputSettings.options['max_frames']
--> 146             mimebundle = fn(element, max_frames=max_frames)
    147             if mimebundle is None:
    148                 return {}, {}

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/ipython/display_hooks.py in layout_display(layout, max_frames)
    218         return None
    219 
--> 220     return render(layout)
    221 
    222 

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/ipython/display_hooks.py in render(obj, **kwargs)
     66         renderer = renderer.instance(fig='png')
     67 
---> 68     return renderer.components(obj, **kwargs)
     69 
     70 

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/renderer.py in components(self, obj, fmt, comm, **kwargs)
    408                 doc = Document()
    409                 with config.set(embed=embed):
--> 410                     model = plot.layout._render_model(doc, comm)
    411                 if embed:
    412                     return render_model(model, comm)

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/panel/viewable.py in _render_model(self, doc, comm)
    453         if comm is None:
    454             comm = state._comm_manager.get_server_comm()
--> 455         model = self.get_root(doc, comm)
    456 
    457         if config.embed:

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/panel/viewable.py in get_root(self, doc, comm, preprocess)
    510         """
    511         doc = init_doc(doc)
--> 512         root = self._get_model(doc, comm=comm)
    513         if preprocess:
    514             self._preprocess(root)

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/panel/layout/base.py in _get_model(self, doc, root, parent, comm)
    120         if root is None:
    121             root = model
--> 122         objects = self._get_objects(model, [], doc, root, comm)
    123         props = dict(self._init_params(), objects=objects)
    124         model.update(**self._process_param_change(props))

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/panel/layout/base.py in _get_objects(self, model, old_objects, doc, root, comm)
    110             else:
    111                 try:
--> 112                     child = pane._get_model(doc, root, model, comm)
    113                 except RerenderError:
    114                     return self._get_objects(model, current_objects[:i], doc, root, comm)

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/panel/pane/holoviews.py in _get_model(self, doc, root, parent, comm)
    237             plot = self.object
    238         else:
--> 239             plot = self._render(doc, comm, root)
    240 
    241         plot.pane = self

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/panel/pane/holoviews.py in _render(self, doc, comm, root)
    304                 kwargs['comm'] = comm
    305 
--> 306         return renderer.get_plot(self.object, **kwargs)
    307 
    308     def _cleanup(self, root):

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/bokeh/renderer.py in get_plot(self_or_cls, obj, doc, renderer, **kwargs)
     71         combining the bokeh model with another plot.
     72         """
---> 73         plot = super(BokehRenderer, self_or_cls).get_plot(obj, doc, renderer, **kwargs)
     74         if plot.document is None:
     75             plot.document = Document() if self_or_cls.notebook_context else curdoc()

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/renderer.py in get_plot(self_or_cls, obj, doc, renderer, comm, **kwargs)
    241             init_key = tuple(v if d is None else d for v, d in
    242                              zip(plot.keys[0], defaults))
--> 243             plot.update(init_key)
    244         else:
    245             plot = obj

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/plot.py in update(self, key)
    989     def update(self, key):
    990         if len(self) == 1 and ((key == 0) or (key == self.keys[0])) and not self.drawn:
--> 991             return self.initialize_plot()
    992         item = self.__getitem__(key)
    993         self.traverse(lambda x: setattr(x, '_updated', True))

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/bokeh/plot.py in initialize_plot(self, plots, ranges)
    907 
    908                 shared_plots = list(passed_plots) if self.shared_axes else None
--> 909                 subplots = subplot.initialize_plot(ranges=ranges, plots=shared_plots)
    910                 nsubplots = len(subplots)
    911 

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/bokeh/plot.py in initialize_plot(self, ranges, plots)
   1051             else:
   1052                 passed_plots = plots + adjoined_plots
-> 1053                 adjoined_plots.append(subplot.initialize_plot(ranges=ranges, plots=passed_plots))
   1054         self.drawn = True
   1055         if not adjoined_plots: adjoined_plots = [None]

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/geoviews/plotting/bokeh/plot.py in initialize_plot(self, ranges, plot, plots, source)
    118                                      overlaid=True, renderer=self.renderer)
    119             shapeplot.geographic = False
--> 120             shapeplot.initialize_plot(plot=fig)
    121         return fig
    122 

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/geoviews/plotting/bokeh/plot.py in initialize_plot(self, ranges, plot, plots, source)
    111     def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
    112         opts = {} if isinstance(self, HvOverlayPlot) else {'source': source}
--> 113         fig = super(GeoPlot, self).initialize_plot(ranges, plot, plots, **opts)
    114         if self.geographic and self.show_bounds and not self.overlaid:
    115             from . import GeoShapePlot

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/bokeh/element.py in initialize_plot(self, ranges, plot, plots, source)
   1396             element = self.hmap.last
   1397         key = util.wrap_tuple(self.hmap.last_key)
-> 1398         ranges = self.compute_ranges(self.hmap, key, ranges)
   1399         self.current_ranges = ranges
   1400         self.current_frame = element

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/plot.py in compute_ranges(self, obj, key, ranges)
    642             if (not (axiswise and not isinstance(obj, HoloMap)) or
    643                 (not framewise and isinstance(obj, HoloMap))):
--> 644                 self._compute_group_range(group, elements, ranges, framewise,
    645                                           axiswise, robust, self.top_level,
    646                                           prev_frame)

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/plotting/plot.py in _compute_group_range(cls, group, elements, ranges, framewise, axiswise, robust, top_level, prev_frame)
    755                     data_range = ds.range(el_dim, dimension_range=False)
    756                 else:
--> 757                     data_range = el.range(el_dim, dimension_range=False)
    758 
    759                 data_ranges[(el, el_dim)] = data_range

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/core/data/__init__.py in pipelined_fn(*args, **kwargs)
    205 
    206             try:
--> 207                 result = method_fn(*args, **kwargs)
    208                 if PipelineMeta.disable:
    209                     return result

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/core/data/__init__.py in range(self, dim, data_range, dimension_range)
    515         elif all(util.isfinite(v) for v in dim.range) and dimension_range:
    516             return dim.range
--> 517         elif dim in self.dimensions() and data_range and bool(self):
    518             lower, upper = self.interface.range(self, dim)
    519         else:

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/core/data/__init__.py in __nonzero__(self)
   1076     def __nonzero__(self):
   1077         "Whether the Dataset contains any values"
-> 1078         return self.interface.nonzero(self)
   1079 
   1080     __bool__ = __nonzero__

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/holoviews/core/data/interface.py in nonzero(cls, dataset)
    496     @classmethod
    497     def nonzero(cls, dataset):
--> 498         return bool(cls.length(dataset))
    499 
    500     @classmethod

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/geoviews/data/geom_dict.py in length(cls, dataset)
    184     @classmethod
    185     def length(cls, dataset):
--> 186         return geom_length(dataset.data['geometry'])
    187 
    188     @classmethod

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/geoviews/util.py in geom_length(geom)
    345     if hasattr(geom, 'exterior'):
    346         geom = geom.exterior
--> 347     if not geom.geom_type.startswith('Multi') and hasattr(geom, 'array_interface_base'):
    348         return len(geom.array_interface_base['data'])//2
    349     else:

~/new/miniconda3/envs/knmi_arctic/lib/python3.8/site-packages/shapely/geometry/base.py in array_interface_base(self)
    324             "removed in Shapely 2.0.",
    325             ShapelyDeprecationWarning, stacklevel=2)
--> 326         return self._array_interface_base()
    327 
    328     @property

TypeError: 'dict' object is not callable
```

</p>
</details>

**Comment by bidhya** [comment]
I get the same error with Shapely 1.8.0. Geoviews was working fine with older version of shapely

**Comment by maximlt** [comment]
Closed by #543 but note that this will only be effective when a new version is released, which should happen soon.
